frontend/src/core/saving/filename.ts

```diff
@@ -5,6 +5,15 @@ import { getFilenameFromDOM } from "../dom/htmlUtils";
 
 const filenameAtom = atom<string | null>(getFilenameFromDOM());
 
+const filetitleAtom = atom((get) => {
```

Comment:
i dont think we need to use a jotai atom for this. this is usually best if you need to hold expensive state or react to updates from it. 

instead we can inline this logic